tests/lib/rules/no-cond-assign.js

```diff
@@ -36,7 +36,8 @@ ruleTester.run("no-cond-assign", rule, {
         "while (someNode || (someNode = parentNode)) { }",
         "do { } while (someNode || (someNode = parentNode));",
         "for (;someNode || (someNode = parentNode););",
-        "if ((function(node) { return (node = parentNode); })(someNode)) { }",
+        { code: "if ((function(node) { return (node = parentNode); })(someNode)) { }", options: ["except-parens"] },
+        { code: "if ((function(node) { return (node = parentNode); })(someNode)) { }", options: ["always"] },
```

Comment:
Moved the rule from invalid to valid in the "always" config, and updated the "except-parens" config to be more explicit, so you can see that the code is now considered valid in either case.